# Darren Hunt

PHYS 434AA

10-22-2020

HW 3

In [1]:
import pandas as pd

# Truth tables

In [2]:
# truth tables
truth1 = [[45,5],[3,47]]
ttable1 = pd.DataFrame(truth1, columns=['True Bird', 'True Person'], index=['NN Bird', 'NN Person'])
ttable1

True Bird  True Person
NN Bird           45            5
NN Person          3           47

In [3]:
truth2 = [[47,11],[1,42]]
ttable2 = pd.DataFrame(truth2, columns=['True Bird', 'True Person'], index=['NA Bird', 'NA Person'])
ttable2

True Bird  True Person
NA Bird           47           11
NA Person          1           42

The above two truth tables display two different algorithms' results for a computerized categorization of bird or person. The first hypothetically applies a neural network NN, and the second algorithm uses some new algorithm NA. The cells True Bird/NN Bird and True Person/NN Person are those correctly sorted by the algorithm. The opposing diagonal are incorrect sorts, where the algorithm incorrectly identifed a bird as a person and a person as a bird.

For the proposed purpose of using either the NN or NA algorithms in an aviary zoo, the ratio of correctly sorted birds and people is relevant. In this specific context, even though either error is able to occur, miscategorizing a person as a bird is the favorable error, meaning it has a less drastic consequence if the algorithm makes this error. We do not want birds to escape (Bird misrecognized as person: True bird/NN or NA person) but a person being held up for a few seconds is fine (Human misrecognized as bird, who: True person NN/ or NA bird).

In [4]:
# mistake percentage
totNN = sum(truth1[0]+truth1[1])
NN_mistake = (((truth1[0][1]) + (truth1[1][0])) / totNN) * 100
print("Percent miscategorized, NN : " + str("{0:.2f}".format(NN_mistake)))

Percent miscategorized, NN : 8.00


In [5]:
totNA = sum(truth2[0]+truth2[1])
NA_mistake = (((truth2[0][1]) + (truth2[1][0])) / totNA) * 100
print("Percent miscategorized, NA : " + str("{0:.2f}".format(NA_mistake)))

Percent miscategorized, NA : 11.88


These simple calculations display the percentage of totally incorrect categorizations by both the neural network and new algorithm. This includes both incorrectly identifying a bird as a person, and a person as a bird. We can see that the neural network algorithm (the first truth table) has an overall less percentage of wrong categorizations. 8 percent of the identifications from this algorithm are wrong, whereas 11.88 percent of the new algorithm's identifications are wrong.

Therefore, the NN neural network makes the fewest mistakes, and the NA new algorithm makes more.

In [6]:
# bird mistaken as person percentage
birdNN =  ((truth1[1][0]) / totNN) * 100
print("Percent of birds falsely identified, NN : " + str("{0:.2f}".format(birdNN)))

birdNA = ((truth2[1][0]) / totNA) * 100
print("Percent of birds falsely identified, NA: " + str("{0:.2f}".format(birdNA)))

Percent of birds falsely identified, NN : 3.00
Percent of birds falsely identified, NA: 0.99


Even though the new algorithm NA has more overall mistakes, the majority of the mistakes made are people misidentified as birds, which is the better option of the two possible mistakes. Since the NA has a smaller percentage of mistaking bird as people, then new algorithm is the better option for this specific scenario for the zoo.

In the scenario that during the pandemic, the only person visiting the aviary zoo is the zoo keeper, and the 52% of photos from the aviary door being people drops to 1%, the truth tables for each algorithm appear as follows:

In [7]:
# if 1% of photos taken are only people (1 out of 100 photos)
newtruth1 = [[96,0],[3,1]]
newttable1 = pd.DataFrame(newtruth1, columns=['True Bird', 'True Person'], index=['NN Bird', 'NN Person'])
newttable1

True Bird  True Person
NN Bird           96            0
NN Person          3            1

In [8]:
newtruth2 = [[98,0],[1,1]]
newttable2 = pd.DataFrame(newtruth2, columns=['True Bird', 'True Person'], index=['NN Bird', 'NN Person'])
newttable2

True Bird  True Person
NN Bird           98            0
NN Person          1            1

If only 1% of the photos are now of people (just the zoo keeper) and this includes both true people and mistakenly identified birds as people, then about 1 out of 100 photos will be be in the lower right cell for both truth tables. Since the amount of photos of people at the door has dropped to be very small, the mistaken number of people as birds is roughly 0 since non-integer identifications here would not make sense. Then, the 1 out of 100 photos being of a person is most likely to be a true person identified as a person by the algorithm. 

(If non-integer numbers were included, this would be split: 0.1 true person/NN bird to 0.9 true person/NN person; and 0.25 true person/NA bird to 0.75 true person/NA person.)

This leaves 99 percent, or 99 out of 100 photos to be of birds, both true birds and birds misidentified as people. Assuming the accuracy of either algorithm has not changed, the percentage of misidentified birds as people would be held constant between the old and new truth tables. Finally, true identified birds holds the majority of the percentage of both truth tables, since most of the photos will be taken of birds in this time.

## Analytic PDFs

**What is the pdf of the sum of two identical exponential distributions?**

To find out the analytic PDF of two identifcal exponential distributions using internet resources, I first start by looking up the based exponential distribution:

https://en.wikipedia.org/wiki/Exponential_distribution

This Wiki page describes the PDf of the exponential distribution. It has a parameter $\lambda$, which is a number greater than 0, where the shape of the distribution as well as statistics such as mean, median, variance, etc. are determined by this lambda value. When we read through the related distributions section, we find there is a long list of related distributions if:

$X\sim Exp(\lambda)$ and $X_i \sim Exp(\lambda_i)$

This is saying we have two exponential distributions, but with different $\lambda$ values. Within this list is the case if $\lambda_i = \lambda$, which is the case we are looking for, where the two exponential distributions are identical. This leads us to a distribution called the Erlang distribution:

https://en.wikipedia.org/wiki/Erlang_distribution

The Erlang distribution has a PDF shape determined by two parameters, *k* (how many exponential distributions we are summing) and $\mu$, the reciprocal of $\lambda$, which we know is an equivalent value between the two added exponential distributions.

Therefore, the PDF of the sum of two identical distributions is the Erlang distribution for ($k=2, \lambda$).

This is also equivalent to a Gamma distribution:

https://en.wikipedia.org/wiki/Gamma_distribution

for values ($k=2, 1/\lambda$).

**What is the pdf of the ratio of two zero-mean unity variance normal distributions  $X_1/X_2$?**

A zero mean unity variance normal distribution is a Gaussian centered around x = 0 with a variance of $\sigma^2 = 1$, which means it also has a standard deviation of $\sigma = 1$ as well. The Wiki page for Normal distributions:

https://en.wikipedia.org/wiki/Normal_distribution#Related_distributions

discusses related distributions to the Gaussian. Upon reading, we find a paragraph on the ratio of two Gaussians if $X_1 , X_2$ are two independent standard normal random variables with mean=0 and variance=1. This leads us to the Cauchy distribution:

https://en.wikipedia.org/wiki/Cauchy_distribution

which is defined on Wiki as the ratio of two independent normal distributions with a mean of 0.

**So far we have looked at 1D probability distributions, but it is possible to have a multi-dimensional vector distribution. A simple first introduction is the 2D Guassian; it looks like a smeared spot. Mathematically this is given by  $X \hat{i}+Y \hat{j}$  where both  X  and  Y  are drawn from 1D Gaussian distributions. If I measure the amplitude of this vector, what is its pdf?**

The amplitude of a vector is given by:

$\lvert V \rvert = \sqrt{(x_2 - x_1)^2 + (y_2 - y_1)^2}$

The Wiki page for a multivariable normal distribution:

https://en.wikipedia.org/wiki/Multivariate_normal_distribution,

discusses that "the PDF of the pdf of the 2-norm (or Euclidean norm) of a multivariate normally distributed vector (centered at zero)" follows a Chi distribution. Therefore, when taking the amplitude of the normal distributions' vectors with a mean of 0, the corresponding PDF is a Chi distribution.

https://en.wikipedia.org/wiki/Chi_distribution